presto-verifier/src/main/java/com/facebook/presto/verifier/Validator.java

```diff
@@ -227,7 +230,28 @@ private boolean validate()
 
         boolean matches = resultsMatch(controlResult, testResult, precision);
         if (!matches && checkDeterministic) {
-            return checkForDeterministicAndRerunTestQueriesIfNeeded();
+            matches = checkForDeterministicAndRerunTestQueriesIfNeeded();
+        }
+        if (matches && !dropTemporaryTables) {
```

Comment:
I think this should be
```java
if (matches && dropTemporaryTables)

Reply:
I think the logic here is that: If checksum succeeds and we didn't drop temporary tables before, we should drop them then.